.cbmc-batch/jobs/aws_cryptosdk_keyring_base_init/aws_cryptosdk_keyring_base_init_harness.c

```diff
@@ -17,12 +17,15 @@
 #include <make_common_data_structures.h>
 
 void aws_cryptosdk_keyring_base_init_harness() {
-    /* Non-deterministic inputs. */
+    /* Nondet input. */
     struct aws_cryptosdk_keyring *keyring   = can_fail_malloc(sizeof(*keyring));
     struct aws_cryptosdk_keyring_vt *vtable = can_fail_malloc(sizeof(*vtable));
 
-    /* Pre-conditions. */
+    /* Assumptions. */
     __CPROVER_assume(keyring != NULL);
```

Comment:
Don't we have a `ensure_cryptosdk_keyring_is_allocated` function? What about an invariant (i.e. `_is_valid` function)? I'd prefer using them here instead of malloc, followed by this assumption. Also, this function name would not even be appropriate anymore, because these predicates should not ensure allocation, but instead nondeterministically allocate memory, so they should be renamed to something like `nondet_allocate_cryptosdk_keyring`.

Reply:
We do have a ensure_cryptosdk_keyring_is_allocated but the function under test (aws_cryptosdk_keyring_base_init) is actually the function that initializes a keyring (including giving it its vtable which is also what the ensure_cryptosdk_keyring_is_allocated does). I changed the can_fail_malloc to bounded_malloc though.